.github/workflows/upd-xna-client.yml

```diff
@@ -0,0 +1,161 @@
+
+        if ($currentVersion -eq $downloadVersion)
+        {
+          echo "Update doesn't required"
+        } else
+        {
+          echo "Update required"
+          
+          # Configure git
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+          # Remove old branch if exist and make new one
+          git branch -D ${{ env.UPDATE_BRANCH_NAME }}
+          git checkout -b ${{ env.UPDATE_BRANCH_NAME }}
+        
+          # Delete old binaries, unpack archive, remove archive
+          rm -r -fo "D:\a\${{ env.REPO_NAME }}\${{ env.REPO_NAME }}\Resources\Binaries"
+          rm -r -fo "D:\a\${{ env.REPO_NAME }}\${{ env.REPO_NAME }}\Resources\BinariesNET8"
+          7z x xna-cncnet-client-*.7z -y -oD:\a\${{ env.REPO_NAME }}\${{ env.REPO_NAME }}
+          rm xna-cncnet-client-*.7z
+
+          # Commit changes
+          git commit -am "${{ env.COMMIT_MESSAGE }}"
+
+          # Path to the updateexec file
+          $updateExecPath = "D:\a\${{ env.REPO_NAME }}\${{ env.REPO_NAME }}\${{ env.CLIENT_PATH }}\updateexec"
+
+          # Check if the updateexec file exists
+          if (-Not (Test-Path -Path $updateExecPath))
+          {
+            echo "The file 'updateexec' does not exist. Skip updating [Delete] section."
+          }
+          else
+          {
+            # Get the deleted and moved files from git diff
+            $deletedFiles = git diff --diff-filter=D --name-status HEAD~1 HEAD | Where-Object { $_.StartsWith('D') } | ForEach-Object { $_.Substring(1).Trim().Replace('/', '\') }
+            $deletedFiles += [Environment]::NewLine
+            $deletedFiles += git diff --diff-filter=R --name-status --diff-filter=R HEAD~1 HEAD | Where-Object { $_.StartsWith('R100') } | ForEach-Object { $_.Substring(4).Trim().Split(' ')[0].Replace('/', '\') }
+
+            # Check if there are any deleted files
+            if ($deletedFiles.Count -eq 0)
+            {
+                echo "No deleted files found in the git diff. Skip updating [Delete] section."
+            }
+            else
+            {
+              # Read the content of the updateexec file
+              $updateexecContent_old = Get-Content -Path $updateExecPath
+              $updateexecContent_new = ""
+
+              # Find the [Delete] section and its position
+              $deleteSectionIndex = $updateexecContent_old.IndexOf("[Delete]")
+
+              if ($deleteSectionIndex -eq -1)
+              {
+                $updateexecContent_old += [Environment]::NewLine 
+                $updateexecContent_old += "[Delete]"
+                $deleteSectionIndex = $updateexecContent_old.IndexOf("[Delete]")
+              }
+
+              # Exclude path from string for config
+              $exclude = "${{ env.CLIENT_PATH }}"
+
+              # Iterate all content of old updateexec
+              foreach($old_line in $updateexecContent_old)
+              {
```

Comment:
Is this how it is done in PowerShell? Seems weird and unnecessary to iterate all content and a bit badly readable, are you sure there's no way of doing this in a simpler way?

https://stackoverflow.com/questions/71148862/how-can-i-convert-a-sed-command-to-its-powershell-equivalent